internal_transfers/actions/src/parse.ts

```diff
@@ -23,32 +20,42 @@ export function partitionEventLogs(logs: Log[]): ClassifiedEvents {
     trades: [],
     settlements: [],
   };
+  const settlementInterface = new ethers.Interface(SETTLEMENT_ABI);
+  const erc20Interface = new ethers.Interface(IERC20_ABI);
   logs.forEach((log, index) => {
-    const topics = log.topics;
-    const eventTopic = topics[0];
-
-    if (eventTopic === TRANSFER_EVENT_TOPIC) {
+    // We are only interested in Transfer Events from erc20 contracts
+    // along with Settlement and Trade Events from the Settlement contract
+    // All other event emissions can be ignored for our purposes.
+    const transferEventLog = erc20Interface.parseLog(log);
+    const settlementEventLog = settlementInterface.parseLog(log);
+    if (transferEventLog != null) {
+      const { from, to } = transferEventLog.args;
       const transfer: TransferEvent = {
-        to: addressFromBytes32(topics[2]),
-        from: addressFromBytes32(topics[1]),
-        amount: BigInt(log?.data),
+        from,
+        to,
+        amount: BigInt(transferEventLog.args[2]),
```

Comment:
Why have you chosen BigInt over https://docs.ethers.org/v5/api/utils/bignumber/? I think you could use ethers bignumber to make some operations simpler and do a little bit less conversions. (but maybe its unrelated to this PR)

Reply:
Port sure big int is a new replacement for that old dirty big number library.